src/network-layer/default/RelayDefaultNetworkLayer.js

```diff
@@ -72,13 +72,22 @@ class RelayDefaultNetworkLayer {
         result => result.json()
       ).then(payload => {
         if (payload.hasOwnProperty('errors')) {
-          var error = new Error(
-            'Server request for query `' + request.getDebugName() + '` ' +
-            'failed for the following reasons:\n\n' +
-            formatRequestErrors(request, payload.errors)
-          );
-          (error: any).source = payload;
-          request.reject(error);
+          if (payload.data && Object.keys(payload.data).length) {
```

Comment:
I originally wrote it this way, but I think payload.data can be `{}`, an empty object. We still need to reject the request in that case I would think